tasks/omnibus.py

```diff
@@ -379,3 +379,26 @@ def manifest(
         omnibus_s3_cache=False,
         log_level=log_level,
     )
+
+@task
+def rpath_edit(ctx, install_path, target_rpath_dd_folder, macos=False):
+    for file in glob.iglob(f"{install_path}/**/*", recursive=True):
+        ext = os.path.splitext(file)[1]
+        if (ext != "" and "so" not in ext and "dylib" not in ext) or "json" in ext or not os.path.isfile(file):
+            continue
+        if macos:
+            toedit_fd = ctx.run(f'otool -l {file} | grep -A 2 "RPATH"', warn=True, hide=True)
+        else:
+            toedit_fd = ctx.run(f'objdump -x {file} | grep "RPATH"', warn=True, hide=True)
+        if install_path in toedit_fd.stdout:
+            new_rpath = os.path.relpath(target_rpath_dd_folder, os.path.dirname(file))
+            if macos:
```

Comment:
This may read somewhat better if we could avoid branching on macos in several places. For example, you could define variables such as `get_rpath` and `patch_rpath` referring to functions that you could assign in a single `if` at the beginning of the function.